tests/e2e/tests/pilot/pod_churn_test.go

```diff
@@ -0,0 +1,281 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package pilot
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/test/application/echo"
+	"istio.io/istio/pkg/test/application/echo/proto"
+	"istio.io/istio/pkg/test/kube"
+)
+
+const (
+	churnAppName         = "churn"
+	churnAppSelector     = "app=" + churnAppName
+	churnAppReplicas     = 10
+	churnAppMinReplicas  = 5
```

Comment:
nit: have some mercy on the single vm circleci. Can you bring this down to 2 or 3?